public/video-ui/src/slices/error.ts

```diff
@@ -0,0 +1,37 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { Action, AnyAction } from 'redux';
+
+const SHOW_ERROR = 'SHOW_ERROR' as const;
+type ShowError = AnyAction & { type: typeof SHOW_ERROR; message: string };
+
+export const showError: (message: string) => ShowError = message => {
+  return {
+    type: SHOW_ERROR,
+    message
+  };
+};
```

Comment:
Redux Toolkit doesn't neatly facilitate custom action types and is opinionated about the format of them `domain/action`: https://redux.js.org/style-guide/#write-action-types-as-domaineventname

Due to how widely this action is used, as a migration step, I am specifying actions in this style with `extraReducers` to handle them. This allows us to step towards proper RTK without making vast changes in one.

Reply:
This would be a bit more long-winded, but you could  define 2x new `reducer`s and actions for setting and clearing the ErrorState with the recommended pattern and type format, but also keep the `extraReducers` to handle the legacy code that is sending "CLEAR_ERROR" and "SHOW_ERROR". 

It might show a clearer path for migrating the legacy code - and we coudl remove the extraReducers when that was done.


Whichever way we go, I'd suggest adding some comments here to explain until we are able to update the other error `dispatch`s to the new pattern.